### DISCUSSION TITLE
Will “wire” work with go generics?

### DISCUSSION THREAD

**Discussion by harishb2k** [google/wire#319] [General]
I know, it is early to ask this question -> Did someone try wire with the current draft implementation for go lang with generics?

**Comment by jay-babu** [comment]
Is there any update on this now that go generics are released?

**Comment by inkognitro** [comment]
It does already? Define a concrete type from a generic one and define its provider as written below.

Concrete type:
```
type ConcreteFoo struct {
    GenericType[Foo]
}
```

The provider:
```
func NewConcreteFoo() *ConcreteFoo {
    return &ConcreteFoo{GenericType[Foo]{}}
}
```

Am I missing something?

edit: Code corrections.

**Comment by stytchiz** [comment]
Wire is currently in maintenance mode (i.e. not accepting new features) and investigation is needed for how much work is involved. Full support of go generics will probably not land anytime soon, but we welcome comments and fixes.

Related: https://github.com/google/wire/issues/354
